### ISSUE TITLE
How to train stabilityai/stable-diffusion-xl-base-1.0 using ImageReward model

### ISSUE THREAD

**Issue by emelpolaris** [THUDM/ImageReward#77]
I found the code snippet for training CompVis/stable-diffusion-v-1-4 using ImageReward.
Based on this, I tried to make the code for training stabilityai/stable-diffusion-xl-base-1.0 but failed.
Is it possible to train stabilityai/stable-diffusion-xl-base-1.0 using ImageReward model?
Thanks

**Comment by xujz18** [comment]
Hello, thanks for your attention! You can refer to our ReFL code and just add the ReFL action similarly in the SDXL fine-tuning script.

**Comment by emelpolaris** [comment]
Thanks for your response.
I tried to add the ReFL action in the SDXL fine-tuning, but no luck.

**Comment by emelpolaris** [comment]
@xujz18 stuck with an error.
could you provide a code for it?

**Comment by emelpolaris** [comment]
If anyone is interested in it, please reach out to me https://join.slack.com/t/xrunnergroup/shared_invite/zt-2h89gocpx-KjI4Vf0Z8ZtA1kP2DvsX1Q

**Comment by amulyaprasanth** [comment]
> Hello, thanks for your attention! You can refer to our ReFL code and just add the ReFL action similarly in the SDXL fine-tuning script.

I was trying the same thing but no luck. Could you please elaborate the steps

**Comment by eaeful** [comment]
The project has added code for ReFL training on StableDiffusion XL
[https://github.com/THUDM/ImageReward/pull/89](https://github.com/THUDM/ImageReward/pull/89)

**Comment by zhangyuting725** [comment]
module load python/3.8
accelerate launch --mixed_precision=bf16 refl_sdxl.py \
  --use_ema \
  --resolution=512 --center_crop --random_flip \
  --train_batch_size=2 \
  --gradient_accumulation_steps=4 \
  --gradient_checkpointing \
  --max_train_steps=100 \
  --learning_rate=1e-05 \
  --image_reward_version ImageReward-v1.0 \
  --max_grad_norm=1 \
  --lr_scheduler="constant" --lr_warmup_steps=0 \
  --output_dir="checkpoint/refl_sdxl" \
  --grad_scale=0.001 \
  --checkpointing_steps 100 \
  --image_base_dir="data/images/" \
  --mapping_batch_size=128 \
  --save_only_one_ckpt \
  --apply_pre_loss \
  --apply_reward_loss
The reward loss and pretrain loss always be nan, is there something wrong or  about the fine-tuning process of  sdxl bf16?
